src/rate_limiter/src/lib.rs

```diff
@@ -508,6 +550,25 @@ pub(crate) mod tests {
         }
     }
 
+    impl PartialEq for BucketReduction {
```

Comment:
Why not let generate this impl ? 
`#[derive(PartialEq)]`

Reply:
I didn't think it can be derived, fixed now.